core-plugins/src/main/java/co/cask/hydrator/plugin/batch/action/WindowsShareCopy.java

```diff
@@ -267,9 +267,23 @@ protected void terminated() {
       this.netBiosSharename = netBiosSharename;
       this.sourceDirectory = sourceDirectory;
       this.destinationDirectory = destinationDirectory;
-      this.bufferSize = (bufferSize == null) && (bufferSize < MIN_BUFFER_SIZE) ? MIN_BUFFER_SIZE : bufferSize;
-      this.numThreads = (numThreads == null) && (numThreads < MIN_NUM_THREADS) ? MIN_NUM_THREADS : numThreads;
+      this.bufferSize = bufferSize;
+      this.numThreads = numThreads;
       this.overwrite = !("false".equals(overwrite));
     }
+
+    public Integer getNumThreads() {
```

Comment:
Also why do all this check every time when you need numthreads and buffersize. The value is not going to change between calls. Just do check before initialization and initialize with right value and use that.